File: MS.NET/Sarvam Backend/Models/MasterResult.cs
using Sarvam.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

public class MasterResult
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("result_id")]
    public int ResultId { get; set; }

    [Required]
    [ForeignKey("MasterModule")]
    [Column("module_id")]
    public int ModuleId { get; set; }

    public MasterModule? MasterModule { get; set; } //  Nullable to avoid validation error

    [Required]
    [ForeignKey("MasterProfile")]
    [Column("PRN")]
    public long PRN { get; set; }

    public MasterProfile? MasterProfile { get; set; } //  Nullable to avoid validation error

    [Range(0, 20)]
    [Column("internals_20")]
    public int? Internals20 { get; set; }

    [Range(0, 40)]
    [Column("lab_40")]
    public int? Lab40 { get; set; }

    [Range(0, 60)]
    [Column("ccee_60")]
    public int? Ccee60 { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/MasterResult.java
package com.cdac.sarvam.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonBackReference;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "master_results")
public class MasterResult {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer resultId;

    @ManyToOne
    @JoinColumn(name = "module_id", nullable = false)
    @JsonBackReference
    private MasterModule module;

    @ManyToOne
    @JoinColumn(name = "prn", referencedColumnName = "PRN", nullable = false) // Correct FK mapping
    private MasterProfile student;

    private Integer internals20;
    private Integer lab40;
    private Integer ccee60;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
}
